controller/proxy-injector/webhook.go

```diff
@@ -131,52 +131,51 @@ func Inject(linkerdNamespace string) webhook.Handler {
 			}, nil
 		}
 
+		// Resource could not be injected with the sidecar, format the reason
+		// for injection being skipped to emit an event
+		var readableReasons string
+		for _, reason := range reasons {
+			readableReasons = readableReasons + ", " + inject.Reasons[reason]
```

Comment:
I know this is just moved from below, but this might be a good opportunity to clean this up by using strings.Join